### PR TITLE
[WIP] Updates to GFF-based data sources

### PR DESCRIPTION
**Pull request by holtgrewe**
See updated `CHANGELOG.md` for changes

I added unit tests but this needs some validation that the result quality does not degrade.

### CODE REVIEW

File: jannovar-core/src/main/java/de/charite/compbio/jannovar/reference/TranscriptModel.java [hunk 1]
```diff
@@ -51,6 +54,13 @@
 	private final String geneID;
 
 	/**
+	 * Alternative gene IDs, as parsed from RefSeq GFF3 file
+	 * 
+	 * See {@link #getAltGeneIDs()} for more information
+	 */
+	private final ImmutableSortedMap<String, String> altGeneIDs;
```
Comment (visze) [inline jannovar-core/src/main/java/de/charite/compbio/jannovar/reference/TranscriptModel.java]: Is the geneID (what ever it is defined) included in the altgeneIDs?

File: jannovar-core/src/main/java/de/charite/compbio/jannovar/reference/TranscriptModel.java [hunk 2]
```diff
@@ -116,6 +137,24 @@ public String getGeneID() {
...
+	 * <dl>
+	 * <dt>GeneID</dt>
+	 * <dd>String of the Entrez/NCBI gene ID</dd>
+	 * <dt>Genbank</dt>
+	 * <dd>String of the Genbank accession</dd>
+	 * <dt>HGNC</dt>
+	 * <dd>String of the numeric HGNC (Human Gene Nomenclature Committee) identifier</dd>
+	 * </dl>
+	 */
+	public ImmutableSortedMap<String, String> getAltGeneIDs() {
```
Comment (visze) [inline jannovar-core/src/main/java/de/charite/compbio/jannovar/reference/TranscriptModel.java]: @holtgrewe So if i want to have the entrez-gene id I have to run this code?

```java
Integer.parseInt(getAltGeneIDs().get("GeneID"))
```

Should we make an enum-set for common ids?

**Comment by visze** [comment]
@holtgrewe For me it seems that you only changed the refSeq parser to use the alternative ids. What about Ensemble, UCSC and CCDS?

File: jannovar-core/src/main/java/de/charite/compbio/jannovar/reference/TranscriptModel.java (hunk 2 above)
Comment (holtgrewe) [inline jannovar-core/src/main/java/de/charite/compbio/jannovar/reference/TranscriptModel.java]: The gene id names are not fixed anywhere.

File: jannovar-core/src/main/java/de/charite/compbio/jannovar/reference/TranscriptModel.java (hunk 1 above)
Comment (holtgrewe) [inline jannovar-core/src/main/java/de/charite/compbio/jannovar/reference/TranscriptModel.java]: Yes.

**Comment by holtgrewe** [comment]
I do not think the other sources provide alternative ids in the files we currently download. Ideally we would download the mapping from hgnc.

The patch needs some testing e.g. for comparison to dbnsfp.

File: jannovar-core/src/main/java/de/charite/compbio/jannovar/reference/TranscriptModel.java (hunk 2 above)
Comment (visze) [inline jannovar-core/src/main/java/de/charite/compbio/jannovar/reference/TranscriptModel.java]: Year. but for example in exomiser we need always entrez-geneIDs.  So we must be sure that the correct name is used in the map to get it.

**Comment by visze** [comment]
dbnsfp?

**Pull request merged**